service/gcs/core/gcs/gcs.go

```diff
@@ -153,13 +154,24 @@ func (c *gcsCore) CreateContainer(id string, settings prot.VMHostedContainerSett
 		return errors.Wrapf(err, "failed to mount layers for container %s", id)
 	}
 
-	// Set up networking.
+	// Stash network adapters away
 	for _, adapter := range settings.NetworkAdapters {
-		if err := c.configureNetworkAdapter(adapter); err != nil {
-			return errors.Wrapf(err, "failed to configure network adapter %s", adapter.AdapterInstanceID)
-		}
 		containerEntry.AddNetworkAdapter(adapter)
 	}
+	// Create the directory that will contain the resolv.conf file.
+	//
+	// TODO(rn): This isn't quite right but works. Basically, when
+	// we do the network config in ExecProcess() the overlay for
+	// the rootfs has already been created. When we then write
+	// /etc/resolv.conf to the base layer it won't show up unless
+	// /etc exists when the overlay is created. This is a bit
+	// problematic as we basically later write to a what is
+	// supposed to be read-only layer in the overlay...  Ideally,
+	// dockerd would pass a runc config with a bind mount for
+	// /etc/resolv.conf like it does on unix.
```

Comment:
Agreed, using a bind mount would be a good move. One thing we still need to look into RE: DNS is that DNS configuration is passed to the GCS on a per-adapter basis. Since resolv.conf is a per-container, rather than per-adapter, configuration solution, if we want to support per-adapter config we'd maybe have to use something like dnsmasq.